mesheryctl/internal/cli/root/config/config.go

```diff
@@ -88,3 +96,69 @@ func (v *Version) GetBuild() string {
 func (v *Version) GetCommitSHA() string {
 	return v.CommitSHA
 }
+
+// SetDefaultVersion sets version of config if not specified
+func (ctx *Context) SetDefaultVersion() error {
+	if ctx.Version == "" {
+		// if no version is specified, pick latest imgaes based on channel
+		if ctx.Channel == "edge" {
+			// if channel is edge, pick the edge-latest images
+			ctx.Version = "latest"
+		} else if ctx.Channel == "stable" {
+			// if the channel is stable, pick the version of the latest stable release
+			version, err := GetLatestStableReleaseTag()
+			ctx.Version = version
+			if err != nil {
+				return errors.Wrapf(err, fmt.Sprintf("failed to fetch latest stable release tag"))
+			}
+		} else {
+			return errors.Errorf("unknown channel %s", ctx.Channel)
+		}
+	}
+	return nil
+}
+
+// GetLatestStableReleaseTag fetches and returns the latest release tag from GitHub
+func GetLatestStableReleaseTag() (string, error) {
+	url := "https://api.github.com/repos/layer5io/meshery/releases/latest"
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to make GET request to %s", url)
+	}
+	defer resp.Body.Close()
+
+	var dat map[string]interface{}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to read response body")
+	}
+	if err := json.Unmarshal(body, &dat); err != nil {
+		return "", errors.Wrap(err, "failed to unmarshal json into object")
+	}
+
+	return dat["tag_name"].(string), nil
+}
+
+// GetCurrentContext returns the current context name and context struct.
+// If the user mentions a temporary context(tempCtxName) with -c flag, change the current-context and proceed to temporary-context
+func GetCurrentContext(tempCtxName string) (string, Context, error) {
```

Comment:
This should be a part of context struct

Reply:
It cannot be part of the context struct because that would be a paradox. The function is there to return context(i.e. the user using it does not have access to context)
it can be part of `MesheryCtlConfig`, but then we will have to create `mctlCfg` everywhere and handle its error. Then we will also have to call the member function and handle its error too. All this will be added everywhere where the only purpose is to get context.